prow/cmd/deck/main.go

```diff
@@ -408,6 +431,171 @@ func handleBadge(ja *jobs.JobAgent) http.HandlerFunc {
 	}
 }
 
+// handleRequestProwJobViews pre-renders a Spyglass page for a Prowjob source
+// A valid Prow job view url takes the form:
+//
+// /view/prowjob/<jobname>/<buildID>
+//
+// Example:
+// - /view/prowjob/echo-test/1021530234601607168
+func handleRequestProwJobViews(sg *spyglass.Spyglass, ca *config.Agent) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		setHeadersNoCaching(w)
+		start := time.Now()
+		src := strings.TrimPrefix(r.URL.Path, "/view/")
+		fmt.Fprint(w, renderSpyglass(sg, ca, src))
+		elapsed := time.Since(start)
+		logrus.Info("Time taken to return Prowjob view page: ", elapsed)
+	}
+}
+
+// handleRequestGCSJobViews pre-renders a Spyglass page for a GCS source
+// A valid job GCS Job view url takes the form:
+//
+// /view/gcs/<bucketname>/<jobprefix>
+//
+// Example:
+// - /view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-large-performance/121
+func handleRequestGCSJobViews(sg *spyglass.Spyglass, ca *config.Agent) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		setHeadersNoCaching(w)
+		start := time.Now()
+		srcData := strings.TrimPrefix(r.URL.Path, "/view/gcs/")
+		src := fmt.Sprintf("gs://%s", srcData)
+		fmt.Fprint(w, renderSpyglass(sg, ca, src))
+		elapsed := time.Since(start)
+		logrus.Info("Time taken to return GCS view page: ", elapsed)
+	}
+}
+
+// handleRequestJobViews handles requests to get all available artifact views for a given job via
+// a general src parameter, which can contain any string used to obtain job artifacts
+// A valid general job view url takes the form:
+//
+// /view/?src=<URI-encoded-source-string>
+//
+// Example:
+// - /view/?src=gs:%2F%2Fkubernetes-jenkins%2Fpr-logs%2Fpull%2Fkubeflow_kubeflow%2F1195%2Fkubeflow-presubmit%2F2558
+func handleRequestJobViews(sg *spyglass.Spyglass, ca *config.Agent) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		setHeadersNoCaching(w)
+		start := time.Now()
+		src := r.URL.Query().Get("src")
+		if src == "" {
+			http.Error(w, "missing src query parameter", http.StatusBadRequest)
+			return
+		}
+
+		fmt.Fprint(w, renderSpyglass(sg, ca, src))
+		elapsed := time.Since(start)
+		logrus.Info("Time taken to return view page: ", elapsed)
```

Comment:
use a log field or (better) prometheus metric

Reply:
metrics will be exported in #8956